docs/en/advanced-concepts/migrating-to-appium-2.0.md

```diff
@@ -127,3 +127,82 @@ TODO
+
+### Suggested capabilities
+
+In addition to the standard `platformName`, `appium:deviceName`, `appium:automationName`, and `appium:platformVersion`, we recommend adopting the following set, where the label `$cloud` is not meant to be interpreted literally but instead should be replaced by your vendor prefix (so for HeadSpin it would be `headspin`, Sauce Labs it would be `sauce`, and BrowserStack it would be `browserstack`, to name just a few examples):
+
+|Capability|Used for|Example|
+|----------|-------|-------|
+|`$cloud:appiumVersion`|Designating which version of the Appium server is used to host and manage drivers. If ommitted, behavior left up to the provider, but the recommendation would be to provide the latest official version.|`2.0.0`|
+|`$cloud:automationVersion`|Designating which version of the specified driver should be used.|`1.55.2`|
+|`$cloud:automation`|Designating a custom driver to use (see below for more info). This would override `appium:automationName` and `$cloud:automationVersion`|`{"name": "org/custom-driver", "source": "github", "package": "custom-driver"}`|
+|`$cloud:appiumPlugins`|Designating the list of plugins (and potentially versions of plugins) to be activated (see below for more info).|`["images", "universal-xml"]`|
+
+### Basic example
+
+Appium extensions (drivers and plugins) have a set of properties that specify where they can be installed from. Cloud providers are obviously under no obligation to provide support for arbitrarily specified extensions, seeing as these may represent untrusted code running in a managed environment. In the case where arbitrary extensions are not supported, the `appium:automationName`, `$cloud:automationVersion`, and `$cloud:appiumPlugins` capabilities should be sufficient. See the following JSON object representing capabilities for a session:
+
+```json
+{
+    "$cloud:appiumVersion": "2.0.0",
+    "platformName": "iOS",
+    "appium:platformVersion": "14.4",
+    "appium:deviceName": "iPhone 11",
+    "appium:app": "Some-App.app.zip",
+    "appium:automationName": "XCUITest",
+    "$cloud:automationVersion": "3.52.0",
+    "$cloud:appiumPlugins": ["images"]
+}
+```
+
+This set of capabilities requests an Appium 2.0 server supporting the XCUITest driver at version `3.52.0`, and the `images` plugin active. This set is easy for a cloud provider to verify. The cloud provider can obviously do anything it wants in response to these capabilities, including downloading Appium and driver and plugin packages on the fly, or erroring out if the versions requested are not in a supported set, or if the plugin is not supported, etc...
+
+### Extension objects
+
+Some service providers may wish to dynamically allow access to all of the features of the Appium 2.0 CLI, including downloading arbitrary drivers and plugins. To represent these extensions, we can define special JSON "extension objects", with the following keys:
```

Comment:
Just want to make sure I understand:

Are these capabilities intended to be provided _before_ the cloud provider starts a server?

Reply:
That would be an implementation detail, but generally speaking, yes, cloud providers don't start internal Appium servers until their front end has received capabilities and processed them and decided which ones to pass on to the internal Appium server, etc...

Reply:
yep ^